registry/storage/catalog_test.go

```diff
@@ -241,85 +240,3 @@ func TestCatalogWalkError(t *testing.T) {
 		t.Errorf("Expected catalog driver list error")
 	}
 }
```

Comment:
We need to add a new test to demonstrate that the bug has been fixed.

Reply:
Would you have any suggestions? I'm not actually familiar with the code base, I just rebased the PR so we can get it merged to move the v3 release forward.

Reply:
Or maybe @milosgajdos can way in here since you worked on the original PR.

Reply:
my memory is fading 😄 but I'll try to have a look

Reply:
Just FYI @deleteriousEffect, that PR _optimises_ the catalog iteration. It does not fix any bugs per se: https://github.com/distribution/distribution/issues/3233#issuecomment-1357731317

The original PR shows benchmarks; I dont think we should keep blocking this PR 🤔

Reply:
> It does not fix any bugs per se

If it doesn't fix any bugs that's alright, but the PR title and commit message needs to reflect that.

Reply:
> the PR title and commit message needs to reflect that.

Fair. I changed the PR title. Updating commit messages is tricky as this is a rebase of the abandoned PR. We could squash them, perhaps and give credit to the original contribution in the squashed commit. WDYT?

Reply:
@milosgajdos I've just changed the commit message so it doesn't mention a bug fix anymore.